src/ImageSharp/Formats/Jpeg/Components/Encoder/HuffmanScanEncoder.cs

```diff
@@ -234,6 +231,65 @@ public void EncodeGrayscale<TPixel>(Image<TPixel> pixels, ref Block8x8F luminanc
             this.FlushInternalBuffer();
         }
 
+        /// <summary>
+        /// Encodes the image with no subsampling and keeps the pixel data as Rgb24.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="pixels">The pixel accessor providing access to the image pixels.</param>
+        /// <param name="luminanceQuantTable">Luminance quantization table provided by the callee.</param>
+        /// <param name="chrominanceQuantTable">Chrominance quantization table provided by the callee.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation.</param>
+        public void EncodeRgb<TPixel>(Image<TPixel> pixels, ref Block8x8F luminanceQuantTable, ref Block8x8F chrominanceQuantTable, CancellationToken cancellationToken)
+            where TPixel : unmanaged, IPixel<TPixel>
+        {
+            this.huffmanTables = HuffmanLut.TheHuffmanLut;
+
+            var unzig = ZigZag.CreateUnzigTable();
+
+            // ReSharper disable once InconsistentNaming
+            int prevDCY = 0, prevDCCb = 0, prevDCCr = 0;
+
+            ImageFrame<TPixel> frame = pixels.Frames.RootFrame;
+            Buffer2D<TPixel> pixelBuffer = frame.PixelBuffer;
+            RowOctet<TPixel> currentRows = default;
+
+            var pixelConverter = new RgbForwardConverter<TPixel>(frame);
+
+            for (int y = 0; y < pixels.Height; y += 8)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                currentRows.Update(pixelBuffer, y);
+
+                for (int x = 0; x < pixels.Width; x += 8)
+                {
+                    pixelConverter.Convert(x, y, ref currentRows);
+
+                    prevDCY = this.WriteBlock(
+                        QuantIndex.Luminance,
+                        prevDCY,
+                        ref pixelConverter.R,
+                        ref luminanceQuantTable,
+                        ref unzig);
```

Comment:
Not sure about actual specs for rgb encoding but I guess we should use same quantization table for all rgb channels. If this is a thing - we should also fix quantization table index for all channels set to either 0 (luminance) or 1 (chrominance).

Reply:
yeah I think you are right. While the image looks correct when decoding with ImageSharp, it looks like other tools like IrfanView interpret the data still as YCbCr. Probably because there are more then one quantization table present.

![output](https://user-images.githubusercontent.com/38701097/129487663-7e9f8fad-f9bd-43c2-842a-17b65b4629a6.jpg)

Reply:
That's strange, decoders should not assume color space via qtable count. Primary 'tag' for rgb should be the app14 marker which you've already implemented and it is working correctly.

What's even funnier - Photoshop opens it correctly.

Reply:
> That's strange, decoders should not assume color space via qtable count. Primary 'tag' for rgb should be the app14 marker which you've already implemented and it is working correctly.

This was just a very wild guess from my side. ImageMagick does also read the image as above and im pretty sure it can handle jpeg with RGB colorspace. If i encode an image with `libjpeg` with RGB colorspace, imagemagick can read it correctly. This means im doing something wrong here, but i can not see at the moment what that is. Any idea's are welcome, I am a bit stuck at the moment.

Reply:
btw. the tiff images which use jpeg compression look correct.

Reply:
I'll take a closer look at the code tomorrow. If photoshop says file is not corrupted - we are somewhat bytes away from cracking it so it shouldn't take long.

Reply:
thank you @br3aker, I appreciate your help.